ios/viewmodels/view-model-example.md

```diff
@@ -0,0 +1,280 @@
+            switch result {
+            case .Success:
+                dispatch_async(dispatch_get_main_queue()) {
+                    self.tableView.reloadData()
+                    self.setupBunnyImage()
+                }
+            case .Failure(let error):
+                // TODO: Need UI here to retry
+                print("Error refreshing preztos: \(error)")
+                return
+            }
+        }
+    }
+
+    private func deletePreztoAtIndex(index: Int) {
+        viewModel.deletePreztoAtIndex(index) { result in
+            dispatch_async(dispatch_get_main_queue()) {
+                switch result {
+                case .Success:
+                    if self.viewModel.count == 0 {
+                        self.goToEmptyVC()
+                    } else {
+                        self.tableView.reloadData()
+                        self.setupBunnyImage()
+                    }
+                case .Failure(let e):
+                    print("Error deleting prezto: \(e)")
+                    self.presentAlertWithTitle("Error", message: "There was an unexpected problem deleting a Prezto.")
+                }
+            }
+        }
+    }
+
+    /*****************************************
+     * Skipping a bunch of helper methods... *
+     *****************************************/
+
+    // MARK: - UITableViewDataSource
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.count + 1
+    }
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 12.0
+    }
+    
+    func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return section == viewModel.count ? 12 : 0
+    }
+    
+    // Snip...
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        if indexPath.section == 0 {
+            if let createPreztoCell = tableView.dequeueReusableCellWithIdentifier(SavedPreztosViewController.CreateNewPreztoTVCIdentifier, forIndexPath: indexPath) as? CreateNewPreztoTableViewCell {
+                createPreztoCell.layer.cornerRadius = 5
+                return createPreztoCell
+            }
+        } else if let savedPreztoCell = tableView.dequeueReusableCellWithIdentifier(SavedPreztosViewController.SavedPreztoTVCIdentifier, forIndexPath: indexPath) as? SavedPreztoTableViewCell {
+            let index = indexPath.section - 1
+
+            savedPreztoCell.delegateVC = self
+            savedPreztoCell.viewModel = viewModel.viewModelForPreztoAtIndex(index)
+            savedPreztoCell.setRightUtilityButtons(createDeleteBtn(), withButtonWidth: 42.0)
+            savedPreztoCell.layer.cornerRadius = 5
+
+            savedPreztoCell.delegate = self
```

Comment:
@paulrolfe Yeah I honestly didn't dig too much into that since this was mostly pre-existing code and I didn't want to break anything.

I like @markdintrepid's idea of removing anything that's not specifically touching the view model so all of this cruft would go away.